libstuff/SQValue.cpp

```diff
@@ -0,0 +1,136 @@
+#include <libstuff/SQValue.h>
+#include <libstuff/sqlite3.h>
+
+// All of our connstructor bodies are emtpy, we just use member initializer lists.
+SQValue::SQValue() : type(SQValue::TYPE::NONE) {}
+SQValue::SQValue(int64_t val) : type(SQValue::TYPE::INTEGER), integer(val) {}
+SQValue::SQValue(double val) : type(SQValue::TYPE::REAL), real(val) {}
+SQValue::SQValue(const char* val) : type(SQValue::TYPE::TEXT), text(val ? val : "") {}
+SQValue::SQValue(const string& val) : type(SQValue::TYPE::TEXT), text(val) {}
+SQValue::SQValue(TYPE t, const string& val) : type(t), text(val) {}
+
+SQValue::operator string() const {
+    switch (type) {
+        case TYPE::TEXT:
+        case TYPE::BLOB:
+            return text;
+        case TYPE::INTEGER:
+            return std::to_string(integer);
+        case TYPE::REAL:
+            // This is the same format that the sqlite3 shell uses internally.
+            char buf[64];
+            sqlite3_snprintf(sizeof(buf), buf, "%!.15g", real);
+            return string(buf);
+        case TYPE::NONE:
+        default:
+            return "";
+    }
+}
+
+string operator+(string lhs, const SQValue& rhs) {
+    lhs += static_cast<string>(rhs);
```

Comment:
Why use += here vs `return lhs + static_cast<string>(rhs)` like below when lhs/rhs are reversed?

Reply:
So this is more efficient than the one below, it avoids an extra copy of the string when passed a temporary value.

The one below then, could look like this one and do:
```
string operator+(SQValue lhs, const string& rhs) {
    string temp = static_cast<string>(lhs)
    temp += rhs;
    return lhs;
}
```

But it no longer avoids the temporary because the cast requires it.